v3/src/components/case-table/index-menu-list.tsx

```diff
@@ -46,12 +65,76 @@ export const IndexMenuList = ({caseId, index}: IProps) => {
       isClosable: true,
     })
   }
+
+  const insertCases = () => {
+    onClose()
+    const casesToAdd = []
+    if (numCasesToInsert) {
+      for (let i=0; i < numCasesToInsert; i++) {
+        casesToAdd.push({})
+      }
+    }
+    data?.addCases(casesToAdd, {[insertPosition]: caseId})
+  }
+  return (
+    <>
+      <MenuList>
+        <MenuItem onClick={handleMoveEntryRow}>Move Data Entry Row Here</MenuItem>
+        <MenuItem onClick={handleInsertCase}>Insert Case</MenuItem>
+        <MenuItem onClick={handleInsertCases}>Insert Cases...</MenuItem>
+        <MenuItem onClick={handleDeleteCase}>Delete Case</MenuItem>
+      </MenuList>
+      <CodapModal
+          isOpen={isOpen}
+          onClose={onClose}
+          title="Insert Cases"
+          hasCloseButton={true}
+          Content={InsertCasesModalContent}
+          contentProps={{numCasesToInsert,
+                          insertPosition,
+                          onChangeNumCasesToInsert: handleNumCasesToInsertChange,
+                          onChangeInsertPosition: handleInsertPositionChange}}
+          buttons={[{ label: "Cancel", onClick: onClose },{ label: "Insert Cases", onClick: insertCases }]}
+      />
+    </>
+
+  )
+}
+
+interface IInsertCasesModalProps {
+  numCasesToInsert: number
+  insertPosition: string
+  onChangeNumCasesToInsert: (value: any) => void
+  onChangeInsertPosition: (value: any) => void
+}
+
+export const InsertCasesModalContent: React.FC<IInsertCasesModalProps> =
```

Comment:
I've been trying to keep our files/modules small and tightly defined. With that in mind, I think as much of the code that specific to the insert cases menu item as possible should be moved into its own file/module.